src/gui/sdlmain.cpp

```diff
@@ -1846,19 +1950,43 @@ static void SetupWindowResolution(const char *val)
 		SDL_Rect bounds;
 		SDL_GetDisplayBounds(sdl.display_number, &bounds);
 		if (w > 0 && h > 0 && w <= bounds.w && h <= bounds.h) {
+			sdl.desktop.want_resizable_window = false;
 			sdl.desktop.window.use_original_size = false;
 			sdl.desktop.window.width = w;
 			sdl.desktop.window.height = h;
 			return;
 		}
 	}
 
-	LOG_MSG("MAIN: 'windowresolution = %s' is not a valid setting, using "
-	        "'original' instead",
-	        pref.c_str());
+        LOG_MSG("MAIN: 'windowresolution = %s' is not a valid setting, using "
+                "'original' instead", pref.c_str());
+        sdl.desktop.want_resizable_window = false;
 	sdl.desktop.window.use_original_size = true;
 }
 
+static SDL_Rect CalculateViewport(int win_width, int win_height)
```

Comment:
Can we use unsigned dimensions here? (otherwise the 'int' implies that negative values might be possible)